tests/users/test_validation_request_data.py

```diff
@@ -212,6 +212,50 @@ def test_user_password_change_request_password_superior_to_limit(self):
 
         self.assertEqual(expected_result, actual_result)
 
+    def test_user_registration_request_data_available_to_mentor_not_boolean(self):
+        expected_result = {"message": "Field need_mentoring must be true or false."}
+        request_body = dict(
+                name=user1['name'],
+                username=user1['username'],
+                password=user1['password'],
+                email=user1['email'],
+                terms_and_conditions_checked=user1['terms_and_conditions_checked'],
+                available_to_mentor='yes')
+        actual_result = validate_user_registration_request_data(request_body)
+
+    def test_user_registration_request_data_available_to_mentor_empty_string(self):
+        expected_result = {"message": "Field need_mentoring must be true or false."}
+        request_body = dict(
+                name=user1['name'],
+                username=user1['username'],
+                password=user1['password'],
+                email=user1['email'],
+                terms_and_conditions_checked=user1['terms_and_conditions_checked'],
+                available_to_mentor='')
+        actual_result = validate_user_registration_request_data(request_body)
+
+    def test_user_registration_request_data_need_mentoring_not_boolean(self):
+        expected_result = {"message": "Field need_mentoring must be true or false."}
+        request_body = dict(
+                name=user1['name'],
+                username=user1['username'],
+                password=user1['password'],
+                email=user1['email'],
+                terms_and_conditions_checked=user1['terms_and_conditions_checked'],
+                need_mentoring='yes')
```

Comment:
When you fix the validation for this field, you won't be needing this test, because these fields are booleans and not strings.